pkg/base/replicated.go

```diff
@@ -137,9 +137,12 @@ func renderReplicated(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (
 			if !parsedBool {
 				continue
 			}
-
-			for k, v := range optionalValues.Values {
-				mergedValues[k] = v
+			if optionalValues.RecursiveMerge {
+				mergedValues = kotsv1beta1.MergeHelmChartValues(kotsHelmChart.Spec.Values, optionalValues.Values)
```

Comment:
`optionalValues` are not always coming from `overlays`, right? i guess i'm confused by the param name `overlayValues` here https://github.com/replicatedhq/kots/pull/1751/files#diff-6f2a6d88643d24717b68bd9924433c1f9f1f4b44567cce0b1b0db8b47486c270R177